src/coreclr/src/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -2,158 +2,536 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
```

Comment:
I think you will need to add these APIs for Mono as well, so that it compiles.

I would split the file into two. Keep the existing methods that are runtime specific as is; and add all new methods under libraries/System.Private.CoreLib.

Reply:
Didn't notice this was a coreclr-specific file. Will fix.